From cb00dc878b0531aa17bbd6f8f5c3f64698f34ea5 Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Mon, 4 Jan 2016 15:33:28 +0800
Subject: [PATCH 1/4] atomisp2: detect exp_id wrap back by 5 frames ahead

Detecting the exp_id wrap back by 5 frames ahead can
unlock the raw buffer in time to avoid FW lack of raw
buffers

Original-Tracked-On: #H1504133802
Change-Id: Icb5ab6a378a04d306bb4d35b26da1f6f328c4046
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/93249
Approver: Jing Tao <jing.tao@intel.com>
Reviewed-by: HaiqiangX Yao <haiqiangx.yao@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: Dessert M BXT RVP (SC)
Build: Dessert M CHT FFD (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-15952
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/459539
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c      | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index e899fd1..1786864 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -6265,7 +6265,7 @@ int atomisp_set_raw_buffer_bitmap(struct atomisp_sub_device *asd, int exp_id)
 		__func__, exp_id, asd->raw_buffer_locked_count);
 
 	/* Check if the raw buffer after next is still locked!!! */
-	exp_id += 2;
+	exp_id += 5;
 	if (exp_id > ATOMISP_MAX_EXP_ID)
 		exp_id -= ATOMISP_MAX_EXP_ID;
 	bitmap = asd->raw_buffer_bitmap + exp_id / 32;
-- 
1.9.1

